fastn-core/src/auth/mod.rs

```diff
@@ -110,3 +139,27 @@ pub async fn get_auth_identities(
     };
     Ok(matched_identities)
 }
+
+async fn set_session_cookie_and_end_response(
+    req: &fastn_core::http::Request,
+    session_id: uuid::Uuid,
+    next: String,
+) -> fastn_core::Result<fastn_core::http::Response> {
```

Comment:
We have added a test framework, check if this code works with it. This code should be fully end to end tested using our test framework.

Reply:
tests using `fastn test` will be added in a separate PR for this auth feature.